### ISSUE TITLE
Integration with Google Home

### ISSUE THREAD

**Issue by podarok** [Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client#253]
**Is your feature request related to a problem? Please describe.**
I tried to connect to my Google Home and failed

**Describe the solution you'd like**
Would be great to have it hooked up to Google Home in order to use laptop for sending commands

**Describe alternatives you've considered**
As an alternative - I should buy a hardware device with Google Home on board

**Additional context**
No context here, except of Google Home can't see this instance of Google Assistant via network

**Comment by andrewdavidmackenzie** [comment]
When I get the assistent working, I can use it (via voice or typed commands) to turn off my lights and such stuff.

But I can't get it to play music on my Google Hub speaker for example.

So, there is some integration there working - but not complete.

**Comment by chaoscreater** [comment]
On the same topic, Routines would be nice to have. You can use Google Home or Google Assistant on phone to run Google Routines, but this isn't possible with the unofficial Google Assistant.

**Comment by lewis1277** [comment]
It would be nice if there were a way with Google home integration to detect when someone with a Google nesr uses the "broadcast" feature, I'm in a family home with nests which we use broadcast to communicate with each other when in different rooms but sometimes miss each others messages due to wearing headphones on pc

If there could be some kind of desktop notification when someone tries it'd be a godsend with home integration

**Comment by chaoscreater** [comment]
> It would be nice if there were a way with Google home integration to detect when someone with a Google nest uses the "broadcast" feature, I'm in a family home with nests which we use broadcast to communicate with each other when in different rooms but sometimes miss each others messages due to wearing headphones on pc
> 
> If there could be some kind of desktop notification when someone tries it'd be a godsend with home integration

That just seems unnecessary on so many levels. If you were located in house A and your family members are in house B, then it would make sense to use the broadcast feature. OR, if you have 2 dwellings on the same land or same property and you and your family members are living in separate dwellings, then it makes sense to use the broadcast feature.

However, if you and all your family members are literally in the same house and literally in the next room, wtf is the point of using broadcast, let alone trying to implement some whacky workaround for when broadcast fails to get the person's attention? Ever thought of just getting up and walking a few steps to start a conversation?

Or maybe adjust the volume on the Google Nest Mini to max volume BEFORE you broadcast? I mean, your entire use case is just whack and someone trying to come up with a workaround for a non-issue is just wasting their efforts. 

I mean, I honestly don't get why you would need a NOTIFICATION for when you MISS a broadcast message. Just send a normal text notification, or normal iMessage or Messenger or whatever notification. You missed the broadcast message already, it's not going to replay itself and if you're going to be using a headphone to block out external sound, then wtf is the point of getting notified of a broadcast message that can't be played back? Sorry, but this whole idea  just sounds stupid.

**Comment by lewis1277** [comment]
> > It would be nice if there were a way with Google home integration to detect when someone with a Google nest uses the "broadcast" feature, I'm in a family home with nests which we use broadcast to communicate with each other when in different rooms but sometimes miss each others messages due to wearing headphones on pc
> > If there could be some kind of desktop notification when someone tries it'd be a godsend with home integration
> 
> That just seems unnecessary on so many levels. If you were located in house A and your family members are in house B, then it would make sense to use the broadcast feature. OR, if you have 2 dwellings on the same land or same property and you and your family members are living in separate dwellings, then it makes sense to use the broadcast feature.
> 
> However, if you and all your family members are literally in the same house and literally in the next room, wtf is the point of using broadcast, let alone trying to implement some whacky workaround for when broadcast fails to get the person's attention? Ever thought of just getting up and walking a few steps to start a conversation?
> 
> Or maybe adjust the volume on the Google Nest Mini to max volume BEFORE you broadcast? I mean, your entire use case is just whack and someone trying to come up with a workaround for a non-issue is just wasting their efforts.
> 
> I mean, I honestly don't get why you would need a NOTIFICATION for when you missed a broadcast message. You missed the broadcast message already, it's not going to replay itself, regardless if you get notified or not. You as the sender of the message should just send a normal text notification, or normal iMessage or Messenger or whatever notification to the other person and let them know what the actual message is. If you're going to be using a headphone to block out external sound, then wtf is the point of getting notified of a broadcast message that can't be played back? Sorry, but this whole idea just sounds stupid.

Fair enough, I've got a house with a disconnected building that I use a lot for work and we use broadcast for laziness/when we're in the middle of something and can't get up but want to let people know of things in the moment. just thought it'd be cool

**Comment by chaoscreater** [comment]
> > > It would be nice if there were a way with Google home integration to detect when someone with a Google nest uses the "broadcast" feature, I'm in a family home with nests which we use broadcast to communicate with each other when in different rooms but sometimes miss each others messages due to wearing headphones on pc
> > > If there could be some kind of desktop notification when someone tries it'd be a godsend with home integration
> > 
> > 
> > That just seems unnecessary on so many levels. If you were located in house A and your family members are in house B, then it would make sense to use the broadcast feature. OR, if you have 2 dwellings on the same land or same property and you and your family members are living in separate dwellings, then it makes sense to use the broadcast feature.
> > However, if you and all your family members are literally in the same house and literally in the next room, wtf is the point of using broadcast, let alone trying to implement some whacky workaround for when broadcast fails to get the person's attention? Ever thought of just getting up and walking a few steps to start a conversation?
> > Or maybe adjust the volume on the Google Nest Mini to max volume BEFORE you broadcast? I mean, your entire use case is just whack and someone trying to come up with a workaround for a non-issue is just wasting their efforts.
> > I mean, I honestly don't get why you would need a NOTIFICATION for when you missed a broadcast message. You missed the broadcast message already, it's not going to replay itself, regardless if you get notified or not. You as the sender of the message should just send a normal text notification, or normal iMessage or Messenger or whatever notification to the other person and let them know what the actual message is. If you're going to be using a headphone to block out external sound, then wtf is the point of getting notified of a broadcast message that can't be played back? Sorry, but this whole idea just sounds stupid.
> 
> Fair enough, I've got a house with a disconnected building that I use a lot for work and we use broadcast for laziness/when we're in the middle of something and can't get up but want to let people know of things in the moment. just thought it'd be cool

That's fine, but at that point all you get is a notification without any context or any actual message. The notification will probably just say "hey you've got a broadcast message just now" and you have no idea what it is. Wouldn't it be easier just to ask Google Assistant to text the other person? Or just send a quick message via Whatsapp or whatever your family use? If this is a legit problem, then it sounds like this has happened enough times already - in which case why not actually just send a notification with proper text message?

Any message I get will obviously show up on my phone, but also on my laptop and so I'll never miss it, regardless if I have headphone with active noise cancellation on or not and regardless if I missed a broadcast message or not. It's a simple solution, don't try to over-engineer or over-complex it.
